crates/ruff_python_ast/generate.py

```diff
@@ -547,6 +566,49 @@ def write_nodekind(out: list[str], ast: Ast) -> None:
     """)
 
 
+# ------------------------------------------------------------------------------
+
+
+def write_node(out: list[str], ast: Ast) -> None:
+    write_node_list = [
+        "ExprBoolOp",
+        "ExprNamed",
+        "ExprBinOp",
+        "ExprUnaryOp",
+        "ExprLambda",
+        "ExprIf",
+        "ExprDict",
+    ]
+    group_names = [group.name for group in ast.groups]
+    node_names = [node.name for node in ast.all_nodes]
+    for group in ast.groups:
+        for node in group.nodes:
+            if node.name not in write_node_list:
+                continue
```

Comment:
I would suggest checking whether the AST node type has a non-`None` `fields` instead of hard-coding a list of nodes here